### PR TITLE
Subscription table migration and api response change.

### PR DESCRIPTION
**Pull request by array-junkie**
Subscription table migration added.

### CODE REVIEW

File: app/Http/Controllers/CampController.php [hunk 1]
```diff
@@ -281,7 +281,8 @@ public function getCampRecord(Request $request, Validate $validate)
                 $camp[] = $livecamp;
                 $indexs = ['topic_num', 'camp_num', 'camp_name', 'key_words', 'camp_about_url', 'nick_name', 'campSubscriptionId'];
                 $camp = $this->resourceProvider->jsonResponse($indexs, $camp);
-                $camp[0]['parentCamps'] = $parentCamp;
+                $camp=$camp[0];
```
Comment (akuks) [inline app/Http/Controllers/CampController.php:284]: Breathing space please :)

File: app/Http/Controllers/CampController.php [hunk 2]
```diff
@@ -757,13 +758,14 @@ public function campSubscription(Request $request, Validate $validate)
...
             $campSubscription->save();
-            $subscriptionId = ($subscription_id) ? "" : $campSubscription->id;
+            $subscriptionId = ($subscription_id && !$all['checked']) ? "" : $campSubscription->id;
             $response = new stdClass();
             $response->msg = $msg;
             $response->subscriptionId = $subscriptionId;
             $indexes = ['msg', 'subscriptionId'];
             $data[0] = $response;
             $data = $this->resourceProvider->jsonResponse($indexes, $data);
+            $data=$data[0];
```
Comment (akuks) [inline app/Http/Controllers/CampController.php:768]: some breathing space please

File: app/Http/Controllers/CampController.php [hunk 3]
```diff
@@ -757,13 +758,14 @@ public function campSubscription(Request $request, Validate $validate)
...
             return $this->resProvider->apiJsonResponse(200, trans('message.success.success'), $data, '');
         } catch (Exception $e) {
             return $this->resProvider->apiJsonResponse(400, trans('message.error.exception'), $e->getMessage(), '');
```
Comment (akuks) [inline app/Http/Controllers/CampController.php:771]: I think `$e->getMessage() ` should be the last not the third parameter

File: app/Http/Controllers/TopicController.php [hunk 4]
```diff
@@ -258,6 +258,7 @@ public function getTopicRecord(Request $request, Validate $validate)
                 $topicRecord[] = $topic;
                 $indexs = ['topic_num', 'camp_num', 'topic_name', 'namespace_name', 'topicSubscriptionId'];
                 $topicRecord = $this->resourceProvider->jsonResponse($indexs, $topicRecord);
+                $topicRecord=$topicRecord[0];
```
Comment (akuks) [inline app/Http/Controllers/TopicController.php:261]: Breathing space :)

**Review by akuks — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/CampController.php (hunk 3 above)
Comment (M-Darab) [reply app/Http/Controllers/CampController.php]: yes. you are right

**Pull request merged**